launch/client.py

```diff
@@ -2373,6 +2380,242 @@ def update_docker_image_batch_job(self, batch_job_id: str, cancel: bool):
+                        else:
+                            post_inference_hooks_strs.append(hook)
+
+                if default_callback_auth_kind is not None:
+                    default_callback_auth = CallbackAuth(
+                        **dict_not_none(
+                            kind=default_callback_auth_kind,
+                            username=default_callback_auth_username,
+                            password=default_callback_auth_password,
+                            cert=default_callback_auth_cert,
+                            key=default_callback_auth_key,
+                        )
+                    )
+                else:
+                    default_callback_auth = None
+
+                payload = dict_not_none(
+                    name=endpoint_name,
+                    model_name=model_name,
+                    source=source,
+                    inference_framework=inference_framework,
+                    inference_framework_image_tag=inference_framework_image_tag,
+                    num_shards=num_shards,
+                    cpus=cpus,
+                    endpoint_type=ModelEndpointType(endpoint_type),
+                    gpus=gpus,
+                    gpu_type=GpuType(gpu_type) if gpu_type is not None else None,
+                    labels=labels or {},
+                    max_workers=max_workers,
+                    memory=memory,
+                    metadata={},
+                    min_workers=min_workers,
+                    per_worker=per_worker,
+                    high_priority=high_priority,
+                    post_inference_hooks=post_inference_hooks_strs,
+                    default_callback_url=default_callback_url,
+                    default_callback_auth=default_callback_auth,
+                    storage=storage,
+                    public_inference=public_inference,
+                )
+                create_model_endpoint_request = CreateLLMModelEndpointV1Request(**payload)
+                response = api_instance.create_model_endpoint_v1_llm_model_endpoints_post(
+                    body=create_model_endpoint_request,
+                    skip_deserialization=True,
+                )
+                resp = json.loads(response.response.data)
+            endpoint_creation_task_id = resp.get("endpoint_creation_task_id", None)  # TODO probably throw on None
+            logger.info("Endpoint creation task id is %s", endpoint_creation_task_id)
+            model_endpoint = ModelEndpoint(
+                name=endpoint_name, bundle_name=f"{endpoint_name}-{str(inference_framework)}"
+            )
+            if endpoint_type == "async":
+                return AsyncEndpoint(model_endpoint=model_endpoint, client=self)
+            elif endpoint_type == "sync":
+                return SyncEndpoint(model_endpoint=model_endpoint, client=self)
+            else:
+                raise ValueError("Endpoint should be one of the types 'sync' or 'async'")
```

Comment:
oops I think this is me but we also should support `StreamingEndpoint` now

Reply:
i think we can keep it this way since we're merging sync and streaming endpoints